python/rtypes/pcc/triggers.py

```diff
@@ -0,0 +1,45 @@
+
+
+class Trigger(object):
+
+    def __init__(self, pcc_type, time, action, priority=None):
+        self.pcc_type = pcc_type
+        self.time = time
+        self.action = action
+        self.priority = priority
+
+    def __call__(self, procedure):
+        class TriggerProcedure(object):
+
+            def __init__(self, procedure):
```

Comment:
since this is being curried at this point, procedure does not need to be passed. It already knows what procedure is.

Reply:
If you want to do it this way, make TriggerProcedure a top level class instead of a nested class. In fact. I think that is better.

Reply:
What does "curried" mean?